File: functions/alsa-lib/variants/go/4_val_seg_5.go
package main

func valSeg(val int32) int {
	r := 1
	val >>= 8
	diff := (val & 0xf0)>>4 + (val & 0x0c)>>2 + (val & 0x02)
	r += int(diff)
	return r
}


func main(){
    valSeg(0)
}

File: functions/alsa-lib/variants/c/4_val_seg_5.c
#include <stdint.h>

static inline int val_seg(int val)
{
	int r = 1;
	val >>= 8;
	if (val & 0xf0) {
		r += 4;
		val >>= 4;
	}
	r += (val & 0x0c) ? (val & 0x0c) >> 2 : 0;
	r += (val & 0x03) ? (val & 0x02) >> 1 : 0;
	return r;
}


int main(){
    val_seg(0);
}
